fairlearn/adversarial/_adversarial_mitigation.py

```diff
@@ -0,0 +1,899 @@
+
+    cuda : str, default = None
+        A string to indicate which device to use when training. For instance,
+        set :code:`cuda='cuda:0'` to train on the first GPU. Only for PyTorch
+        backend.
+
+    warm_start : bool, default = False
+        Normally, when set to False, a call to fit triggers reinitialization,
+        which discards the models and intializes them again. Setting to
+        True triggers reuse of these models. Note: if pre-initialized models
+        were passed, the model (and its parameters) are never discard.
+
+    random_state : int, RandomState instance, default = None
+        Controls the randomized aspects of this algorithm, such as shuffling.
+        Useful to get reproducible output across multiple function calls.
+
+    References
+    ----------
+    .. [1] Zhang, Lemoine, Mitchell `"Mitigating Unwanted Biases with
+       Adversarial Learning" <https://dl.acm.org/doi/pdf/10.1145/3278721.3278779>`_,
+       AIES, 2018.
+
+    """  # noqa : E501
+
+    def __init__(
+        self,
+        *,
+        backend="auto",
+        predictor_model=[],  # [] is a NN with no hidden layers.
+        adversary_model=[],
+        predictor_loss="auto",
+        adversary_loss="auto",
+        predictor_function="auto",
+        threshold_value=0.5,
+        predictor_optimizer="Adam",
+        adversary_optimizer="Adam",
+        constraints="demographic_parity",
+        y_transform="auto",
+        sf_transform="auto",
+        learning_rate=0.001,
+        alpha=1.0,
+        epochs=1,
+        batch_size=-1,
+        max_iter=-1,
+        shuffle=False,
+        shuffle_start=False,
+        progress_updates=None,
+        skip_validation=False,
+        callbacks=None,
+        cuda=None,
+        warm_start=False,
+        random_state=None,
+    ):
+        """Initialize class by only storing (kw)args, as per sklearn API."""
+        self.backend = backend
+        self.predictor_model = predictor_model
+        self.adversary_model = adversary_model
+        self.predictor_loss = predictor_loss
+        self.adversary_loss = adversary_loss
+        self.predictor_function = predictor_function
+        self.threshold_value = threshold_value
+        self.predictor_optimizer = predictor_optimizer
+        self.adversary_optimizer = adversary_optimizer
+        self.constraints = constraints
+        self.y_transform = y_transform
+        self.sf_transform = sf_transform
+        self.learning_rate = learning_rate
+        self.alpha = alpha
+        self.epochs = epochs
+        self.batch_size = batch_size
+        self.max_iter = max_iter
+        self.shuffle = shuffle
+        self.shuffle_start = shuffle_start
+        self.progress_updates = progress_updates
+        self.skip_validation = skip_validation
+        self.callbacks = callbacks
+        self.cuda = cuda
+        self.warm_start = warm_start
+        self.random_state = random_state
+
+    def __setup(self, X, Y, A):
```

Comment:
```suggestion
    def _setup(self, X, Y, A):
```

the canonical way to indicate a method is private is a single `_`.